### PR TITLE
Fix react testing conflicts from Reaction

### PR DESCRIPTION
**Pull request by eessex**
- Updates test libs related to react
- Adds some config to `test.config.js` to handle relay setup
- Fixes tests broken by relay hooks

### CODE REVIEW

File: test.config.js [hunk 1]
```diff
@@ -43,3 +45,7 @@ try {
 require("raf/polyfill")
 require("should")
 require("./src/lib/jade_hook")
+
+// Used by relay
+sd.METAPHYSICS_ENDPOINT = "http://localhost:5003"
```
Comment (damassi) [inline test.config.js]: Interesting that we haven't tested a relay-based component yet in force

File: test.config.js [hunk 2]
```diff
@@ -43,3 +45,7 @@ try {
...
+global.Promise = require("promise")
```
Comment (damassi) [inline test.config.js]: Why is this needed?

File: test.config.js (hunk 1 above)
Comment (zephraph) [reply test.config.js]: Ohhh, that's interesting. I was wondering why this was needed, but I hadn't thought of it from that angle.

File: test.config.js (hunk 2 above)
Comment (eessex) [reply test.config.js]: To catch the error caused by metaphysics not connecting to its fake url-- there is some conflict in jest's implementation of promise, this is a frequently recommended solution to avoid using it.

Comment (damassi) [reply test.config.js]: Mind adding a comment to that issue? It seems odd that we're globally overwriting a mature (native) implementation of something with a library.

File: test.config.js (hunk 1 above)
Comment (eessex) [reply test.config.js]: We actually have been, but the new hooks setup in `SystemContextProvider` seemed to cause an issue, we have tests that were x'd 3 months back to handle the same problem.

**Comment by zephraph** [comment]
Only tangentially related, but we should probably also update react: https://github.com/artsy/force/pull/4398

File: test.config.js (hunk 2 above)
Comment (eessex) [reply test.config.js]: Sure just added a link to the docs - https://jestjs.io/docs/en/troubleshooting#unresolved-promises

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `dangerfile.ts`.
  </td>
    </tr>
  </tbody>
</table>



## Error FetchError
```
invalid json response body at https://metaphysics-staging.artsy.net/ reason: Unexpected token < in JSON at position 0
FetchError: invalid json response body at https://metaphysics-staging.artsy.net/ reason: Unexpected token < in JSON at position 0
    at /home/circleci/project/node_modules/isomorphic-fetch/node_modules/node-fetch/lib/body.js:48:31
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
### Dangerfile
```
---------------------------------^
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2646dc425b52f83fc1363c3ea51ddde914680a15
</p>
